### DISCUSSION TITLE
How to disable eslint complexity rule for functional component name

### DISCUSSION THREAD

**Discussion by khushbooPareek** [eslint/eslint#17025] [Q&A] [answered]
Hi,
In my application I am using functional component of react. and I have added eslint rule 'complexity": ["error", 20]' rule.
It is showing me error for the main component level function also. But I just want it for inner functions. 
![image](https://user-images.githubusercontent.com/35484944/227928166-3a08ff4c-887f-4083-a1f8-c045a8228d6a.png)

Please let me know if there is a way to do this.
Thanks.

**Comment by mdjermanovic** [comment, marked as answer]
Hi @khushbooPareek!

> Please let me know if there is a way to do this.

No, this rule doesn't provide options to define exceptions. You could use [`eslint-disable`](https://eslint.org/docs/latest/use/configure/rules#disabling-rules) comments (for example, `// eslint-disable-next-line complexity` right before the function you want to allow), or copy the core rule into a custom rule and make it behave the way you want.
